### PR TITLE
feat: Add sqlite read/write pool split

### PR DESCRIPTION
**Pull request by dav1do**
Linear ticket WS1-1431.

For now, put a basic struct in ceramic_core that manages access to the pools. The sqlx details are not hidden and still relied upon by crates interacting with the database, as the `sqlx::SqlitePool` is part of the public interface.

### CODE REVIEW

File: core/src/sql.rs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+impl SqlitePool {
+    /// Connect to the sqlite database at the given path. Creates the database if it does not exist.
+    /// Uses WAL journal mode.
+    pub async fn connect(path: impl AsRef<Path>) -> anyhow::Result<Self> {
+        let db_path = format!("sqlite:{}", path.as_ref().display());
+        let conn_opts = SqliteConnectOptions::from_str(&db_path)?
+            .journal_mode(SqliteJournalMode::Wal)
+            .create_if_missing(true)
+            // .synchronous(sqlx::sqlite::SqliteSynchronous::Normal) // normally enough in WAL mode?
```
Comment (dav1do) [inline core/src/sql.rs:21]: Opinions on this setting? I also assume optimize on close is okay for both the RO and W pools.

File: core/src/sql.rs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            .optimize_on_close(true, None);
+
+        let ro_opts = conn_opts.clone().read_only(true);
+
+        let writer = SqlitePoolOptions::new()
+            .max_connections(1)
+            .connect_with(conn_opts)
+            .await?;
+        let reader = SqlitePoolOptions::new()
+            .max_connections(32) //TODO
```
Comment (dav1do) [inline core/src/sql.rs:31]: How configurable do we want the pool to be? we could accept ConnectionOptions (or something similar if we want a lot of control), but it seems like RO connections might be the only value we change for now.

Comment (nathanielc) [reply core/src/sql.rs:31]: I am fine without much configuration until we discover we need it.

**Review by nathanielc — APPROVED** [review]
LGTM

**Pull request merged**